python/ray/_raylet.pyx

```diff
@@ -1433,41 +1434,29 @@ async def execute_streaming_generator_async(
     executor = worker.core_worker.get_event_loop_executor()
     interrupt_signal_event = threading.Event()
 
-    futures = []
     try:
         try:
             async for output in gen:
-                # NOTE: Reporting generator output in a streaming fashion,
-                #       is done in a standalone thread-pool fully *asynchronously*
-                #       to avoid blocking the event-loop and allow it to *concurrently*
-                #       make progress, since serializing and actual RPC I/O is done
-                #       with "nogil".
-                futures.append(
-                    loop.run_in_executor(
-                        executor,
-                        report_streaming_generator_output,
-                        context,
-                        output,
-                        cur_generator_index,
-                        interrupt_signal_event,
-                    )
-                )
-                cur_generator_index += 1
-        except Exception as e:
-            # Report the exception to the owner of the task.
-            futures.append(
-                loop.run_in_executor(
+                # NOTE: Reporting generator output in a streaming fashion, is
+                # done in a standalone thread-pool to avoid blocking the
+                # event-loop , since serializing and actual RPC I/O is done
+                # with "nogil". We still wait for the report to finish to
+                # ensure that the task does not modify the output before we
+                # serialize it. Note that the RPC is sent asynchronously, and
+                # we do not wait for the reply here (unless we are under
+                # backpressure).
```

Comment:
Please clarify that back-pressure doesn't apply to RPC Actor calls (only Ray Data)